client/landing/stepper/declarative-flow/internals/steps-repository/readymade-template-generate-content/index.tsx

```diff
@@ -1,23 +1,138 @@
+import DocumentHead from 'calypso/components/data/document-head';
+import FormattedHeader from 'calypso/components/formatted-header';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormTextarea from 'calypso/components/forms/form-textarea';
+import { Step } from 'calypso/landing/stepper/declarative-flow/internals/types';
+import { ONBOARD_STORE, SITE_STORE } from 'calypso/landing/stepper/stores';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { ReadymadeTemplate } from 'calypso/my-sites/patterns/types';
+import generateAIContentForTemplate from './api/generate-content';
+import ReadymadeTemplatePreview from './components/readymade-template-preview';
+import TextProgressBar from './components/text-progress-bar';
+import type { OnboardSelect } from '@automattic/data-stores';
+import './style.scss';
+
+type ReadymadeTemplateGenerateContentProps = {
+	readymadeTemplate: ReadymadeTemplate;
+	siteId: number;
+	siteSlug: string;
+};
+
+const ReadymadeTemplateGenerateContent: React.FC< ReadymadeTemplateGenerateContentProps > = ( {
+	readymadeTemplate,
+	siteId,
+	siteSlug,
+} ) => {
+	const [ aiContext, setAiContext ] = useState( '' );
+	const [ numberOfGenerations, setNumberOfGenerations ] = useState( 0 );
+	const [ isGeneratingContent, setIsGeneratingContent ] = useState( false );
+	const { assembleSite } = useDispatch( SITE_STORE );
+
+	const handleTextareaChange = ( event: { target: { value: string } } ) => {
+		setAiContext( event.target.value );
+	};
+
+	const generateContent = () => {
+		setIsGeneratingContent( true );
+		setNumberOfGenerations( numberOfGenerations + 1 );
+		generateAIContentForTemplate( readymadeTemplate, aiContext )
+			.then( ( rt: ReadymadeTemplate ) =>
+				assembleSite( siteId, 'assembler', {
+					homeHtml: rt.home.content,
+					headerHtml: rt.home.header,
+					footerHtml: rt.home.footer,
+					pages: [],
+					globalStyles: rt.globalStyles,
+					canReplaceContent: true,
+					siteSetupOption: 'readymade-template',
+				} )
+			)
+			.then( () => setIsGeneratingContent( false ) );
+	};
 
-const ReadymadeTemplateGenerateContent = () => {
 	return (
-		<div>
-			<h1>Generate Content with Me</h1>
-			<p>Launchpad things here for reasons</p>
-		</div>
+		<>
+			<div className="generate-content">
+				<DocumentHead title={ translate( 'Customize Content With AI' ) } />
+				<form className="generate-content__form">
+					<FormFieldset className="generate-content__form-fieldset">
+						<FormLabel htmlFor="tagline">{ translate( 'Describe your site' ) }</FormLabel>
+						<p className="generate-content__description">
+							{ translate(
+								`Describe your site in a few sentences. The more details you give us the better results you'll get.`
+							) }
+						</p>
+						<FormTextarea
+							name="tagline"
+							id="tagline"
+							placeholder={ translate( 'Write an amazing description of your site.' ) }
```

Comment:
Would an example of an amazing description be useful here? Just so people are better aware of the things that might be worth mentioning and what kind of length is useful?

The BigSky research (pekYwv-40F-p2) found that people usually enter weak and poorly described prompts, we should seek to help them write a better one.

Reply:
Any ideas/suggestions?

Reply:
I think maybe it depends upon whether the user is building a personal blog, a small business website and so on. We could infer that from the RT chosen, but to keep things simple how about just hardcoding something like: 

> The Beachcomber Bistro is a cafe offering amazing food, delicious coffee and local beers. It's located next to the beach at Harlyn Bay, offering a stunning view from our deck. We serve breakfasts in the morning then pizza, burgers and a changing array of specials from lunch until sunset.

I'm sure there are better versions but the above seems like something someone with no experience of AI could easily write about their own business.

Reply:
Updated